### PR TITLE
Adding section on conda to the contributing guide.

### LINKED ISSUE #1620: Delete FeatureLabs evalml-core-feedstock fork, add conda update dev guide
Conda-forge asks that any PR to update the [evalml feedstock](https://github.com/conda-forge/evalml-core-feedstock) comes from [a personal fork](https://github.com/dsherry/evalml-core-feedstock), meaning one that you create individually using your github account, rather than using a fork owned by our organization. [The reason for this](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork), I believe, is that forks in GitHub which are owned by orgs cannot give the "Allow edits from maintainers" permission, but forks owned by individuals can.

I'm filing this to ask @freddyaboulton : is there any reason we should keep the FeatureLabs fork of the evalml feedstock, or can we delete it to avoid confusion? I'd like to delete it :)

Also, this tracks writing / updating a process doc somewhere in the evalml repo to describe what is required to make changes to the conda feedstock. This could go into [the `contributing` guide](https://github.com/alteryx/evalml/blob/main/contributing.md), or we could find a different place.

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #1620 

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Review by chukarsten — APPROVED** [review]
Can this be audited without changing anything?  It looks good as is.  But if it can, I can run through the steps if it'd be helpful.

File: contributing.md [hunk 1]
```diff
@@ -86,6 +86,25 @@ If your work includes a [breaking change](https://en.wiktionary.org/wiki/breakin
     * Description of your breaking change
 ```
 
+### 4. Updating our conda package
+
+Some pull requests, in particular those that add new dependencies, may require editing our conda [package](https://anaconda.org/conda-forge/evalml).
```
Comment (angela97lin) [inline contributing.md:91]: update or add new dependencies? :d

**Review by angela97lin — APPROVED** [review]
Looks good! 🚢

File: contributing.md (hunk 1 above)
Comment (freddyaboulton) [reply contributing.md:91]: Good call!

**Comment by freddyaboulton** [comment]
> Can this be audited without changing anything? It looks good as is. But if it can, I can run through the steps if it'd be helpful.

@chukarsten You can try adding yourself to the list of maintainers by running through the steps! Let me know if we should make any changes hehe

**Comment by chukarsten** [comment]
Don't wait on me @freddyaboulton

**Review by dsherry — COMMENTED** [review]
This is great! 🐍 📖

**Pull request merged**